gallery/gallery/lib/studies/fortnightly/shared.dart

```diff
@@ -0,0 +1,532 @@
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+class ArticleData {
+  ArticleData({this.imageUrl, this.category, this.title, this.snippet});
+
+  final String imageUrl;
+  final String category;
+  final String title;
+  final String snippet;
+}
+
+class HorizontalArticlePreview extends StatelessWidget {
+  HorizontalArticlePreview({this.data, this.minutes});
+
+  final ArticleData data;
+  final int minutes;
+
+  @override
+  Widget build(BuildContext context) {
+    TextTheme textTheme = Theme.of(context).textTheme;
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Expanded(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(
+                data.category,
+                style: textTheme.subhead,
+              ),
+              SizedBox(height: 12),
+              Text(
+                data.title,
+                style: textTheme.headline.copyWith(fontSize: 16),
+              ),
+            ],
+          ),
+        ),
+        if (minutes != null) ...[
+          Text(
+            GalleryLocalizations.of(context).craneMinutes(minutes),
+            style: textTheme.body2,
+          ),
+          SizedBox(width: 8)
+        ],
+        Image.asset(
+          data.imageUrl,
+          fit: BoxFit.cover,
+          excludeFromSemantics: true,
+        ),
+      ],
+    );
+  }
+}
+
+class VerticalArticlePreview extends StatelessWidget {
+  VerticalArticlePreview({
+    this.data,
+    this.width,
+    this.headlineTextStyle,
+    this.showSnippet = false,
+  });
+
+  final ArticleData data;
+  final double width;
+  final TextStyle headlineTextStyle;
+  final bool showSnippet;
+
+  @override
+  Widget build(BuildContext context) {
+    TextTheme textTheme = Theme.of(context).textTheme;
+
+    return SizedBox(
+      width: width ?? double.infinity,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          SizedBox(
+            width: double.infinity,
+            child: Image.asset(
+              data.imageUrl,
+              fit: BoxFit.fitWidth,
+              excludeFromSemantics: true,
+            ),
+          ),
+          SizedBox(height: 12),
+          Text(
+            data.category,
+            style: textTheme.subhead,
+          ),
+          SizedBox(height: 12),
+          Text(
+            data.title,
+            style: headlineTextStyle ?? textTheme.headline,
+          ),
+          if (showSnippet) ...[
+            SizedBox(height: 4),
+            Text(
+              data.snippet,
+              style: textTheme.body1,
+            ),
+          ],
+        ],
+      ),
+    );
+  }
+}
+
+List<Widget> buildArticlePreviewItems(BuildContext context) {
```

Comment:
This is used to create a list to feed into a ListView widget so, it would require wrapping it in a column in order to be a standalone widget. Also wouldn't be able to easily wrap each item with padding for the mobile layout. I think I'll leave as-is.